### PR TITLE
fix: [SRM-12329]: Add portal for GroupName (#11758)

### PR DESCRIPTION
**Pull request by praneshg239**
* fix: [SRM-12329]: Add portal for GroupName

* fix: [SRM-12329]: Updated common group name component and fix issues in the component

* fix: [SRM-12329]: Update tests

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

#### [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

[SRM-12329]: https://harness.atlassian.net/browse/SRM-12329?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SRM-12329]: https://harness.atlassian.net/browse/SRM-12329?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SRM-12329]: https://harness.atlassian.net/browse/SRM-12329?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/modules/85-cv/components/GroupName/GroupName.tsx [hunk 1]
```diff
@@ -6,18 +6,43 @@
...
+    usePortal
+  } = props
   const { getString } = useStrings()
   const addNewOption = { label: getString('cv.addNew'), value: '' }
 
+  const dropdownOptions: SelectOption[] = groupNames
+
+  // Adding "+ Add New" option based on  "allowAddGroup" prop
+  if (allowAddGroup) {
+    const isAddNewOptionPresent = groupNames.find(groupName => groupName.label === addNewOption.label)
```
Comment (harshilbits) [inline src/modules/85-cv/components/GroupName/GroupName.tsx:38]: what is groupNames is undefined or empty array ?

File: src/modules/85-cv/components/GroupName/GroupName.tsx [hunk 2]
```diff
@@ -52,16 +77,20 @@ export default function GroupName(props: GroupNameProps): JSX.Element {
...
+      usePortal={usePortal}
+      selectProps={selectProps}
       label={label ?? getString('cv.monitoringSources.prometheus.groupName')}
       disabled={disabled}
       value={item}
-      name={fieldName}
+      name={dropdownFieldName}
       items={groupNames || []}
       onChange={selectedItem => {
         if (selectedItem?.label === addNewOption.label) {
```
Comment (harshilbits) [inline src/modules/85-cv/components/GroupName/GroupName.tsx:88]: Null check for `addNewOption.label`

Comment (praneshg239) [reply src/modules/85-cv/components/GroupName/GroupName.tsx:88]: it is a constant

File: src/modules/85-cv/components/GroupName/GroupName.tsx (hunk 1 above)
Comment (praneshg239) [reply src/modules/85-cv/components/GroupName/GroupName.tsx]: We have assigned `groupNames` to empty array `[ ]` by default. So it will work.

**Review by deepesh-ui — DISMISSED** [review]

**Review by harshilbits — APPROVED** [review]

**Review by deepesh-ui — APPROVED** [review]

**Review by vkbansal-harness — APPROVED** [review]

**Pull request merged**